### ISSUE TITLE
Regression in version 1.2.0: selectAll does not return complete char(32) strings

### ISSUE THREAD

**Issue by cwangfr** [hipercompl/fbdb#7]
**Description:** After upgrading from fbdb version 1.1.1 to version 1.2.0, I encountered a new problem. In version 1.1.1, selectAll would correctly return the entire char(32) string from my database. However, in version 1.2.0, it only returns the first 8 characters, truncating the rest of the data.

**Steps to Reproduce:**
Create a table with a char(32) field.
Insert a 32-character string into this field.
Use selectAll to retrieve the record.
Observe that only the first 8 characters of the string are returned in version 1.2.0.

**Expected Behavior:** All 32 characters should be returned, as they were in version 1.1.1.

**Actual Behavior:** Only the first 8 characters of the string are returned. The remaining characters are missing.

**Additional Information:**

Database version: 2.5.8
Database character set: NONE
Flutter version: 3.19.6
**fbdb version:**
Works correctly in 1.1.1
Issue present in 1.2.0

ver 1.1.1
<img width="771" alt="8734912b6d6e828da6583965fbddd40" src="https://github.com/user-attachments/assets/5b4609c0-dbeb-419d-b22f-bdf44f100e0e" />

ver 1.2.0
<img width="775" alt="91f431003539cdd3c7c84650e5936f4" src="https://github.com/user-attachments/assets/492bdf96-626b-49aa-956a-b423cfafd57d" />

DB.select function
<img width="485" alt="6f874987e20edae664c9bda068284c4" src="https://github.com/user-attachments/assets/fca1274d-ec42-4101-b219-2fd751e2095a" />

**Comment by ttyrakow** [comment]
I'll investigate the issue. That's definitely a regression related to issue #5.

**Comment by ttyrakow** [comment]
Yes, unfortunately the method used to resolve #5 works fine with databases encoded in UTF-8, but doesn't work for single-byte encoded databases / fields. Working on a solution, will post an updated version shortly.

*Edit:* the method worked fine for all database encodings **except** `NONE`. Neither Unicode encodings nor single-byte encodings caused problems. `NONE` did.

**Comment by ttyrakow** [comment]
Just pushed 1.2.1 to pub.dev for publishing. Added special test cases for databases with character set `NONE`, right now all tests pass, both for `UTF-8` and `NONE` database character sets.
Thanks for reporting and sorry for inconvenience - the `CHAR` fields in Firebird are sort of pain in the butt from the client code's point of view. `VARCHAR`s are much nicer to work with.

The method used to resolve this issue isn't 100% foolproof. If you happen to have a `CHAR` field containing a value **intentionally** right-padded with spaces (and not auto-padded by the DB), the padding still may get truncated.

**Comment by ttyrakow** [comment]
Please check if 1.2.1 works well in your scenarios, so that the issue can be closed.

**Comment by ttyrakow** [comment]
As a workaround, to make sure no intentional padding gets truncated client-side, you can do `select cast(CHAR_32_FIELD as varchar(32)) as CHAR_32_FIELD from ...` instead of just `select CHAR_32_FIELD from ...`. This way the client code will receive **exact** length of the returned text (which will be a `VARCHAR` instead of `CHAR`) and no truncating will occur. Just letting you know in case you _do_ have intentionally right-padded strings in `CHAR` fields of a `NONE`-encoded database.

**Comment by ttyrakow** [comment]
OK, I've implemented a better solution IMO, which depends on the actual character set of the returned value. For character sets `NONE`, `ASCII` and `OCTETS` there is not truncation, for all other character sets the text is truncated to (byte count / 4) characters. Added some specific test cases to guard against another regression.
Version 1.2.2 has just been published.
This solution should handle properly padded `CHAR(N)` texts, which are actually shorter than `N` characters and are right-padded to the required length `N` by the database.

**Comment by ttyrakow** [comment]
Fixed in version 1.2.1, then refined the fix in 1.3.1, to trim the `CHAR(N)` values to `N` **runes** rather than UTF-16 code points (which might have caused the string to be over-trimmed, if it contained emojis or other characters out of Unicode plane 0).
